File: snippets/syntax/SyntaxTest.cpp
// Sample C++ file for testing VSCode theme syntax highlighting
#include <iostream>
#include <vector>
#include <map>
#include <string>
#include <memory>
#include <functional>

#define MAX_COUNT 100
const std::string GREETING = "Hello, C++!";

// Forward declaration
class SyntaxTest;

/**
 * @brief A sample class to demonstrate syntax highlighting
 */
class SyntaxTest {
private:
    std::string name;
    int count;
    std::vector<std::string> items;

public:
    // Constructor
    explicit SyntaxTest(const std::string& name) : name(name), count(0) {}

    // Destructor
    ~SyntaxTest() = default;

    // Template function
    template<typename T>
    T add(T a, T b) {
        return a + b;
    }

    // Method with various syntax elements
    void performActions() {
        // Local variables
        int number = 42;
        double pi = 3.14159;
        bool isActive = true;

        // Pointers and references
        int* ptr = &number;
        std::string& ref = name;

        // Control structures
        if (number > 0) {
            std::cout << GREETING << " " << ref << std::endl;
        } else if (number == 0) {
            std::cout << "Zero detected" << std::endl;
        } else {
            std::cout << "Negative number" << std::endl;
        }

        // Loop
        for (int i = 0; i < MAX_COUNT; ++i) {
            items.push_back("Item " + std::to_string(i));
        }

        // Switch
        enum class Color { Red, Green, Blue };
        Color color = Color::Blue;
        switch (color) {
            case Color::Red:
                std::cout << "Color is red" << std::endl;
                break;
            case Color::Green:
                std::cout << "Color is green" << std::endl;
                break;
            case Color::Blue:
                std::cout << "Color is blue" << std::endl;
                break;
        }

        // Lambda expression
        auto printer = [](const std::string& s) { std::cout << "Lambda: " << s << std::endl; };
        for (const auto& item : items) {
            printer(item);
        }

        // Exception handling
        try {
            if (number == 0) throw std::runtime_error("Division by zero");
            double result = 10.0 / number;
            std::cout << "Result: " << result << std::endl;
        } catch (const std::exception& e) {
            std::cerr << "Error: " << e.what() << std::endl;
        }

        // Smart pointers and collections
        auto shared = std::make_shared<std::string>("Shared");
        std::map<std::string, int> scores;
        scores["Alice"] = 95;
        scores["Bob"] = 85;
    }

    // Inline method
    inline std::string getName() const { return name; }
};

// Operator overloading
std::ostream& operator<<(std::ostream& os, const SyntaxTest& obj) {
    os << "SyntaxTest(name: " << obj.getName() << ")";
    return os;
}

// Template class
template<typename T>
class Container {
    T value;
public:
    explicit Container(T val) : value(val) {}
    T getValue() const { return value; }
};

// Macro
#define LOG(msg) std::cout << "LOG: " << msg << std::endl

// Main function
int main() {
    SyntaxTest test("TestUser");
    test.performActions();
    LOG("Macro test");

    // Template usage
    Container<int> container(42);
    std::cout << "Container value: " << container.getValue() << std::endl;

    // Function pointer
    void (*funcPtr)() = []() { std::cout << "Function pointer called" << std::endl; };
    funcPtr();

    return 0;
}

File: snippets/syntax/SyntaxTest.java
// Sample Java file for testing VSCode theme syntax highlighting
package syntax;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;
import java.util.function.Consumer;

/**
 * A sample class to demonstrate syntax highlighting.
 * @author YourName
 */
public class SyntaxTest {
    // Constants
    private static final String GREETING = "Hello, World!";
    private static final int MAX_COUNT = 100;

    // Instance variables
    private String name;
    private int count;
    private List<String> items = new ArrayList<>();

    // Enum
    enum Color {
        RED, GREEN, BLUE
    }

    /**
     * Constructor
     * @param name The name to initialize
     */
    public SyntaxTest(String name) {
        this.name = name;
        this.count = 0;
    }

    // Method with various syntax elements
    public void performActions() {
        // Local variables
        int number = 42;
        double pi = 3.14159;
        boolean isActive = true;

        // Control structures
        if (number > 0) {
            System.out.println(GREETING + " " + name);
        } else if (number == 0) {
            System.out.println("Zero detected");
        } else {
            System.out.println("Negative number");
        }

        // Loop
        for (int i = 0; i < MAX_COUNT; i++) {
            items.add("Item " + i);
        }

        // Switch with enum
        Color color = Color.BLUE;
        switch (color) {
            case RED:
                System.out.println("Color is red");
                break;
            case GREEN:
                System.out.println("Color is green");
                break;
            case BLUE:
                System.out.println("Color is blue");
                break;
            default:
                System.out.println("Unknown color");
        }

        // Lambda expression
        Consumer<String> printer = (s) -> System.out.println(s);
        items.forEach(printer);

        // Try-catch
        try {
            int result = 10 / number;
            System.out.println("Result: " + result);
        } catch (ArithmeticException e) {
            System.err.println("Error: " + e.getMessage());
        } finally {
            System.out.println("Cleanup complete");
        }

        // Generics and collections
        Map<String, Integer> scores = new HashMap<>();
        scores.put("Alice", 95);
        scores.put("Bob", 85);
    }

    // Getter with annotation
    @Deprecated
    public String getName() {
        return name;
    }

    // Main method
    public static void main(String[] args) {
        SyntaxTest test = new SyntaxTest("TestUser");
        test.performActions();
    }
}